clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinatorTest.java

```diff
@@ -143,6 +145,88 @@ private void setupCoordinator(int retryBackoffMs, int rebalanceTimeoutMs, Option
                                                 mockTime);
     }
 
+    @Test
+    public void testMetrics() {
+        setupCoordinator();
+
+        metrics.sensor("heartbeat-latency").record(1.0d);
+        metrics.sensor("heartbeat-latency").record(2.0d);
+        metrics.sensor("heartbeat-latency").record(6.0d);
+
+        assertNotNull(getMetric("heartbeat-response-time-max"));
+        assertNotNull(getMetric("heartbeat-rate"));
+        assertNotNull(getMetric("heartbeat-total"));
+        assertEquals(6.0d, getMetric("heartbeat-response-time-max").metricValue());
+        assertEquals(0.1d, getMetric("heartbeat-rate").metricValue());
```

Comment:
Why is rate `0.1d` ?

Reply:
This is dependent on the window size: the rate value is calculated by the current count and the elapsed time proportional to the window size.